File: itmo/M4236/lapko/hw2_lock_free_priority_queue/src/main/java/priority_q/PriorityQueue.java
package priority_q;

import java.util.Queue;

/**
 * Lock-Free очередь с приоритетами
 *
 * @param <E extends Comparable<E>> Тип элементов
 */
public interface PriorityQueue<E extends Comparable<E>> extends Queue<E> {

    /**
     * Проверка очереди на пустоту
     * <p>
     * Метод должен быть lock-free (wait-free для уверенных в себе)
     *
     * @return true если очередь пуста, иначе - false
     */
    boolean isEmpty();
}

File: itmo/M4236/kuznetsov/hw2/src/main/kotlin/ru/itmo/hpc/PriorityQueue.kt
package ru.itmo.hpc

import java.util.*

/**
 * Lock-Free очередь с приоритетами
 * @param <E> Тип элементов
 */
interface PriorityQueue<E: Comparable<E>> : Queue<E> {

    /**
     * Проверка очереди на пустоту
     *
     * Метод должен быть lock-free (wait-free для уверенных в себе)
     *
     * @return true если очередь пуста, иначе - false
     */
    override fun isEmpty(): Boolean
}